installers/custom_ipxe/main.go

```diff
@@ -1,24 +1,96 @@
 package custom_ipxe
 
 import (
+	"encoding/json"
 	"strings"
 
+	"github.com/packethost/pkg/log"
+	"github.com/tinkerbell/boots/installers"
 	"github.com/tinkerbell/boots/ipxe"
 	"github.com/tinkerbell/boots/job"
 )
 
 func init() {
-	job.RegisterSlug("custom_ipxe", bootScript)
+	job.RegisterInstaller("ipxe", ipxeScript)
+	job.RegisterSlug("custom_ipxe", ipxeScript)
 }
 
-func bootScript(j job.Job, s *ipxe.Script) {
+func ipxeScript(j job.Job, s *ipxe.Script) {
+	logger := installers.Logger("ipxe")
+	if j.InstanceID() != "" {
+		logger = logger.With("instance.id", j.InstanceID())
+	}
+
+	var cfg *Config
+	var err error
+
+	if j.OperatingSystem().Installer == "ipxe" {
+		cfg, err = ipxeConfigFromJob(j)
+		if err != nil {
+			s.Echo("Failed to decode installer data")
+			s.Shell()
+			logger.Error(err, "decoding installer data")
+
+			return
+		}
+	} else {
+		cfg = &Config{}
```

Comment:
don't think this is needed, is it?

Reply:
`cfg` at this point would be nil. But I can create it during the setting of the value instead of before.